spec/engine/matchers/part_of_set_matcher_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe SplitIoClient::PartOfSetMatcher do
+  subject { SplitIoClient::SplitFactory.new('', { logger: Logger.new('/dev/null') }).client }
+
+  let(:split_json) { File.read(File.expand_path(File.join(File.dirname(__FILE__), '../../test_data/splits/part_of_set_matcher/split.json'))) }
+
+  before do
+    stub_request(:get, 'https://sdk.split.io/api/splitChanges?since=-1')
+      .to_return(status: 200, body: split_json)
+  end
+
+  it 'validates the treatment is ON for correct attribute value' do
+    expect(subject.get_treatment('user', 'LOCAL_IS_PART_OF', %w(a b c))).to eq SplitIoClient::Treatments::ON
```

Comment:
Do other matchers test are done this way with JSON? doesn't it become more testeable if you test the matcher in isolation? like array (input, operation, output) =>  asesertion?

create a test with empty array, the matcher should be false.

and one with null/nil just to see if you get control